### PR TITLE
Require less alignment for attn bias (#114173)

### PR DESCRIPTION
**Pull request by drisspg**
Improved Fix for Attention Mask Alignment Issue (#112577)

This PR addresses Issue #112577 by refining the previously implemented fix, which was found to be incorrect and causes un-needed memory regressions. The update simplifies the approach to handling the alignment of the attention mask for mem eff attention.

Alignment Check and Padding: Initially, the alignment of the attention mask is checked. If misalignment is detected, padding is applied, followed by slicing. During this process, a warning is raised to alert users.

Should this be warn_once?

We only call expand, once on the aligned mask.

Reference
https://github.com/facebookresearch/xformers/blob/main/xformers/ops/fmha/cutlass.py#L115

@albanD, @mruberry, @jbschlosser, @walterddr, and @mikaylagawarecki.

Pull Request resolved: https://github.com/pytorch/pytorch/pull/114173
Approved by: https://github.com/danthe3rd

Fixes #ISSUE_NUMBER

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

### CODE REVIEW

File: test/inductor/test_torchinductor.py [hunk 1]
```diff
@@ -6441,6 +6441,52 @@ def fn_or(x, y):
             (torch.randn(32), torch.randn(32)),
         )
 
+    @requires_cuda()
+    @unittest.skipIf(
+        not PLATFORM_SUPPORTS_FUSED_SDPA,
```
Comment (atalman) [inline test/inductor/test_torchinductor.py:6446]: On main PR this is called: PLATFORM_SUPPORTS_MEM_EFF_ATTENTION, why on cherry pick we name this as : PLATFORM_SUPPORTS_FUSED_SDPA

**Comment by drisspg** [comment]
Merge in this into this patch:
https://github.com/pytorch/pytorch/pull/114909

**Review by atalman — APPROVED** [review]

**Review by huydhn — APPROVED** [review]
